inginious/frontend/plugins/custom_input/__init__.py

```diff
@@ -0,0 +1,58 @@
+import os
+import web
+import json
+
+from inginious.frontend.plugins.utils import create_static_resource_page
+from inginious.client.client_sync import ClientSync
+from inginious.frontend.pages.api._api_page import APIAuthenticatedPage
+from inginious.frontend.parsable_text import ParsableText
+
+_static_folder_path = os.path.join(os.path.dirname(__file__), "static")
+
+def customInputManagerWithCurriedClient(client):
+    class CustomInputManager(APIAuthenticatedPage):
+        def __init__(self):
+            self._client = client
+
+        def add_unsaved_job(self, task, inputdata):
+            temp_client = ClientSync(self._client)
+            return temp_client.new_job(task, inputdata)
+
+        def API_POST(self):
+            request_params = web.input()
+            courseid = request_params.get("courseid")
```

Comment:
Use the new function in utilities `get_mandatory_parameter` to retrieve the parameters.